integration-tests/source/IntegrationTestResourceHandler.cpp

```diff
@@ -323,3 +342,155 @@ std::string IntegrationTestResourceHandler::GetResourceId(const std::string &res
 {
     return resource.substr(resource.find('/'));
 }
+vector<ActiveViolation> IntegrationTestResourceHandler::GetViolations(const std::string &profileName)
+{
+    vector<ActiveViolation> violations;
+
+    ListActiveViolationsRequest request;
+    request.SetSecurityProfileName(profileName);
+
+    ListActiveViolationsOutcome outcome = iotClient.ListActiveViolations(request);
+
+    if (!outcome.IsSuccess())
+    {
+        printf(
+            "Failed to List Active Violations for: %s\n%s\n",
+            profileName.c_str(),
+            outcome.GetError().GetMessage().c_str());
+    }
+    else
+    {
+        violations = outcome.GetResult().GetActiveViolations();
+    }
+    if (violations.empty())
+    {
+        printf("Found no violations for Security Profile: %s\n", profileName.c_str());
+    }
+    return violations;
+}
+void IntegrationTestResourceHandler::CreateAndAttachSecurityProfile(
+    const string &profileName,
+    const string &thingGroupName,
+    const vector<std::string> &metrics)
+{
+    MetricValue metricValue;
+    metricValue.SetCount(1L);
+
+    BehaviorCriteria criteria;
+    criteria.SetComparisonOperator(ComparisonOperator::less_than);
+    criteria.SetDurationSeconds(300);
+    criteria.SetValue(metricValue);
+    criteria.SetConsecutiveDatapointsToAlarm(1);
+    criteria.SetConsecutiveDatapointsToClear(1);
```

Comment:
do you think this line can be causing any issue? would probably be safer to set this to 100 or something

Reply:
I hope that is the issue. Just changed! Thanks!